### PR TITLE
Fix missing buffer termination call and add a guard for corruption of…

### PR DESCRIPTION
**Pull request by atris**
… buffer

### CODE REVIEW

**Comment by hlinnaka** [comment]
An Assert is not appropriate here. PQExpBufferIsBroken returns true if you have ran out of memory, which can happen. Should check it with regular if(), and report an OOM error up to the user.

On 28 March 2016 14:54:32 EEST, Atri Sharma <notifications@github.com> wrote:
>… buffer
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/greenplum-db/gpdb/pull/578
>
>-- Commit Summary --
>
>* Fix missing buffer termination call and add a guard for corruption of
>buffer
>
>-- File Changes --
>
>    M src/backend/cdb/cdbconn.c (5)
>
>-- Patch Links --
>
>https://github.com/greenplum-db/gpdb/pull/578.patch
>https://github.com/greenplum-db/gpdb/pull/578.diff
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/greenplum-db/gpdb/pull/578

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by atris** [comment]
Thanks, fixed

File: src/backend/cdb/cdbconn.c [hunk 1]
```diff
@@ -381,6 +381,13 @@ cdbconn_doConnect(SegmentDatabaseDescriptor *segdbDesc,
 
 	Assert (nkeywords < MAX_KEYWORDS);
 
+		if (PQExpBufferDataBroken(buffer))
+		{
+			write_log("Out Of Memory encountered %s\n",
+			 buffer.data);
```
Comment (hlinnaka) [inline src/backend/cdb/cdbconn.c]: No point in printing buffer.data here. It's always going to be just "out of memory".

File: src/backend/cdb/cdbconn.c [hunk 2]
```diff
@@ -381,6 +381,13 @@ cdbconn_doConnect(SegmentDatabaseDescriptor *segdbDesc,
...
+			 
+			raise(SIGABRT);
```
Comment (hlinnaka) [inline src/backend/cdb/cdbconn.c]: Why do you want to bring down the system on OOM? It's a "normal" error condition, we should handle it, return an error to the user, and continue.

Comment (vraghavan78) [inline src/backend/cdb/cdbconn.c]: +1

Comment (atris) [inline src/backend/cdb/cdbconn.c]: Fixed,thanks

**Comment by hlinnaka** [comment]
Actually, the 'buffer' isn't used for anything, so it should be just removed.

**Comment by atris** [comment]
Closed as we remove buffer from above function now. A separate PR for the same has been made.

**Pull request closed without merging**